SU2_CFD/src/SU2_CFD.cpp

```diff
@@ -118,17 +118,30 @@ int main(int argc, char *argv[]) {
       if (nZone > 1 ) {
         SU2_MPI::Error("The required solver doesn't support multizone simulations", CURRENT_FUNCTION);
       }
-      if (config->GetDiscrete_Adjoint())
+      if (config->GetDiscrete_Adjoint()) {
+        if (config->GetMultiphysicsDiscrete_Adjoint())
```

Comment:
> I am guessing these special cases will be removed once all of the drivers have settled into place?

They definitely could be removed, even in this PR (the `if (config->GetMultiphysicsDiscrete_Adjoint())` in fact just determines which AD-approach is used - right now that's only the case for `CDiscAdjMultizoneDriver`. 
However it feels to me that this would mean to many changes at once on such a basic level.
It would also require some performance tests, as the current one for single zone problems should be faster as there is just one `std::vector` instead of a large amount of dereferencings down to `CVariable`. It's very likely that this improves a lot with #753

> Which leads to the larger question: do you expect that all multiphysics problems can now be adapted to this driver?

Yes that's the overall idea behind coming up with this generic framework instead of a discrete adjoint CHT driver. Though a lot of work still has to be done to cover all with one and the same driver.
Fluid-fluid interface problems should be an easy round, FSI will be a joint work with @pcarruscag 

> Lots of potential

:+1:

Reply:
Just a follow-up note as what I wrote here

> as the current one for single zone problems should be faster as there is just one std::vector instead of a large amount of dereferencings down to CVariable.

is nonsense, as in both cases we first go to `CVariable` (and then we read indices from an external structure in the standard or from an own one in added one) so there shouldn't be any difference in performance.

@talbring just encouraged me to change it directly everywhere, I'll consider it.

Reply:
However adaptions would then probably also have to be made to `CDiscAdjFEASolver` to not break the adjoint FSI cases. Maybe we should really postpone it to the FSI integration into the multiphysics driver, that would line up well.